gravity-forms/gw-choice-counter.php

```diff
@@ -113,34 +113,59 @@ function output_script() {
-							if ( ! values ) {
-								// If no values provided in the config, just get the number of checkboxes checked.
-								if ( self.isCheckableField( $choiceField ) ) {
-									count += $choiceField.find( ':checked' ).not(' #choice_' + choiceFieldIds[ i ] + '_select_all').length;
+						// Prevent count field from being recalculated if it's hidden.
+						if ( ! gformIsHidden( countField ) ) {
+							for ( var i = 0; i < choiceFieldIds.length; i++ ) {
+
+								var $choiceField = $( '#input_' + formId + '_' + choiceFieldIds[ i ] );
+								if ( ! values ) {
+									// If no values provided in the config, just get the number of checkboxes checked.
+									if ( self.isCheckableField( $choiceField ) ) {
+										count += $choiceField.find( ':checked' ).not(' #choice_' + choiceFieldIds[ i ] + '_select_all').length;
+									} else {
+										count += $choiceField.find( 'option:selected' ).length;
+									}
 								} else {
-									count += $choiceField.find( 'option:selected' ).length;
+									// When values are provided, match the values before adding them to count.
+									var selectedValues = [];
+									$choiceField.find( ':checked' ).each( function( k, $selectedChoice ) {
+										selectedValues.push( $selectedChoice.value );
+									});
+									values.forEach( function( val ) {
+										count += selectedValues.indexOf( val ) >= 0;
+									});
 								}
-							} else {
-								// When values are provided, match the values before adding them to count.
-								var selectedValues = [];
-								$choiceField.find( ':checked' ).each( function( k, $selectedChoice ) {
-									selectedValues.push( $selectedChoice.value );
-								});
-								values.forEach( function( val ) {
-									count += selectedValues.indexOf( val ) >= 0;
-								});
+
 							}
 
+							if( parseInt( countField.val() ) != parseInt( count ) ) {
+								countField.val( count ).change();
+							}
 						}
 
-						if( parseInt( countField.val() ) != parseInt( count ) ) {
-							countField.val( count ).change();
+					};
+
+					// Prevent count field to reset to default if hidden by conditional logic.
+					gform.addFilter('gform_reset_pre_conditional_logic_field_action', function (reset, formId, targetId, defaultValues, isInit) {
+						var id = targetId.split( '_' ).pop();
+						if ( id == args.countFieldId ) { 
+							return false;
 						}
 
-					};
+						return reset;
+					});
+
+					// Tackle show/hide action to count field. Recalculation when show is triggered.
+					gform.addAction('gform_post_conditional_logic_field_action', function (formId, action, targetId, defaultValues, isInit) {
+						var id = targetId.split( '_' ).pop();
+						if ( id == args.countFieldId ) { 
+							if ( action == 'hide' ) {
+								$( targetId ).find( 'input' ).val('0').trigger( 'change' );
+							} else if ( action == 'show' ) {
+								self.updateChoiceEventHandler();
+							}
+						}
+					});
 
```

Comment:
My instinct is to simplify this code to this:

```js
// Recalculate Count field when it is revealed by conditional logic.
gform.addAction( 'gform_post_conditional_logic_field_action', function ( formId, action, targetId ) {
	var id = targetId.split( '_' ).pop();
	if ( id == args.countFieldId && action === 'show' ) {
		self.updateChoiceEventHandler();
	}
} );
```

We don't necessarily need the Count field to be `0` when hidden via conditional logic and since we haven't encountered a use-case that requires it, letting GF clear the value when hidden works well.

Reply:
@spivurno skipping the '0' update on the counter field doesn't completely work. Here is a quick 40s video of how it is after the suggested update: https://www.loom.com/share/fe4ebb34e33f4580b264670899f591ec
​
​Lmk what you think!

Reply:
@spivurno Updated!